File: storage/storage-mysql/src/main/java/io/say/better/storage/mysql/dao/repository/MemberReadRepository.java
package io.say.better.storage.mysql.dao.repository;

import io.say.better.core.enums.Provider;

import java.util.Optional;

public interface MemberReadRepository<T> {
	Optional<T> findByProviderAndLoginId(Provider provider, String loginId);
	Optional<T> findByEmail(String email);
}

File: storage/storage-mysql-kt/src/main/kotlin/io/say/better/storage/mysql/dao/repository/MemberReadRepository.kt
package io.say.better.storage.mysql.dao.repository

import io.say.better.core.enums.Provider
import io.say.better.storage.mysql.domain.entity.Member
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional

interface MemberReadRepository : JpaRepository<Member, Long> {
    fun findByProviderAndLoginId(
        provider: Provider,
        loginId: String,
    ): Optional<Member>

    fun findByEmail(email: String): Optional<Member>
}
